### DISCUSSION TITLE
Plex App API config assumes exposure of TCP 32400

### DISCUSSION THREAD

**Discussion by jehanalvani** [linuxserver/Heimdall#740] [App issues]
When configuring the enhanced Plex app, Heimdall assumes that TCP 32400 is exposed to communicate with a Plex server. In many cases, folks will deploy a server, and expose it on 80/443 via a reverse proxy, so 32400 is never exposed to the internet. As the API is REST, there's no reason that we should assume 32400. 

Example is visible below in my own config. I had to expose 32400 without proxying via my reverse-proxy, then configure a custom URL specifying `http` and `:32400` to successfully connect to the PMS API. 

![Screen Shot 2021-12-09 at 1 19 20 PM copy](https://user-images.githubusercontent.com/2472689/145478219-98cf0bdc-0fb9-4bad-8ebb-c0c3721cf517.png)

**Comment by hobodave** [comment]
Can work around this by simply explicitly providing 443 port in the URL. e.g. `https://plex.local:443/`

**Reply by spider1163** [reply]
I use `http://plex:32400` on Docker

**Comment by zeroquinc** [comment]
For me it doesnt show anything? It says that it communicated with the API but I see no info.
Any idea's?

![msedge_KjhgdFwTZn](https://user-images.githubusercontent.com/39315068/181718694-814d8b1d-c191-40aa-9b56-44736a33f31a.png)

![uVS0FCBQFI](https://user-images.githubusercontent.com/39315068/181718711-f0e6a660-f9c0-43df-bab6-0ec6f8e5825a.png)

**Reply by pvries86** [reply]
I have the same issue, did you find a fix for this?

**Reply by zeroquinc** [reply]
Sadly not yet, if I find one I'll let you know.

**Reply by pbarone** [reply]
Same for me... successful test but only see the plain tile

**Reply by Dallas-J** [reply]
I was having this same issue - it looks like the SSL verification toggle broke the plugin.

You can fix it by copying https://github.com/linuxserver/Heimdall-Apps/blob/10c81dbc5f0d532dabc736c354cb24174655358f/Plex/Plex.php to `<HEIMDALL_CONFIG>/www/SupportedApps/Plex/Plex.php` on your local machine for the time being.

P.S. I also opened an issue: https://github.com/linuxserver/Heimdall-Apps/issues/540

**Reply by pbarone** [reply]
That fixed it for me, nice

**Comment by jehanalvani** [comment]
My solution was in my original Issue note: 

>  I had to expose 32400 without proxying via my reverse-proxy, then configure a custom URL specifying http and :32400 to successfully connect to the PMS API.

HTH

**Reply by pvries86** [reply]
Yes, I got that far, but I'm having the same issue as @zeroquinc. The communication with the API is successful, but the icon doesn't show any extra info.

**Reply by zeroquinc** [reply]
Would love to hear if someone finds a fix, I still have the issue too. Without reverse or with reverse proxy, it doesn't matter in my case. It can always connect to the API but on the dashboard there is nothing, just the Plex logo.

**Comment by bdelieson** [comment]
After replacing my Plex.php with the link from Dallas-J, still not working for me.  API communication successful but no info on the dashboard.  I have other Heimdall icons with working API's.  I am running Plex v4.87.2, Heimdall 2.5.4, http://ip:32400, no proxy.
